src/ir/operation.cc

```diff
@@ -23,22 +23,43 @@
 
 namespace raksha::ir {
 
+// Note: I tried to make this only defined in terms of the type parameter T, but
+// the compiler was unhappy with me writing std::function<std::string(const T&)>
+// and I could not figure out how to placate it.
+template<class T, class U>
+static std::string PrintNamedMapInNameOrder(
+    const absl::flat_hash_map<std::string, T> &map_to_print,
+    U value_printing_fn) {
+  std::vector<absl::string_view> names;
+  names.reserve(map_to_print.size());
+  for (auto &key_value_pair : map_to_print) {
+    names.push_back(key_value_pair.first);
+  }
+  std::sort(names.begin(), names.end());
+  return absl::StrJoin(
+      names, ", ",
+      [&](std::string* out, const absl::string_view name) {
+        auto find_res = map_to_print.find(name);
+        CHECK(find_res != map_to_print.end())
+          << "Name " << name << " not found in map_to_print";
```

Comment:
You could simply use `map_to_print.at(name)` as it is guaranteed to exist?

Reply:
I did not know that `at` existed on `unordered_map`! Thanks for pointing that out.

Reply:
Done.